File: java/src/com/jhobot/commands/admin/CurrentThreads.java
package com.jhobot.commands.admin;

import com.jhobot.handle.MessageHandler;
import com.jhobot.handle.Util;
import com.jhobot.handle.commands.*;
import sx.blah.discord.handle.impl.events.guild.channel.message.MessageReceivedEvent;
import sx.blah.discord.util.EmbedBuilder;

import java.awt.*;
import java.util.HashMap;
import java.util.List;
import java.util.Random;

public class CurrentThreads implements Command {
    @Override
    public Runnable run(MessageReceivedEvent e, List<String> args) {
        return () -> {
            MessageHandler m = new MessageHandler(e.getChannel());
            EmbedBuilder b = new EmbedBuilder();
            b.withFooterText(Util.getTimeStamp());

            b.withColor(new Color(new Random().nextFloat(), new Random().nextFloat(), new Random().nextFloat()));
            b.withTitle("Current Threads Running");
            ThreadCountHandler.HANDLER.getMap().forEach((k, v) -> b.appendField(k.getName() + " [" + k.getLongID() + "]", Integer.toString(v.size()), false));
            m.sendEmbed(b.build());
        };
    }

    @Override
    public Runnable help(MessageReceivedEvent e, List<String> args) {
        HashMap<String, String> st = new HashMap<>();
        st.put("threads", "Displays all running threads for users.");
        return HelpHandler.helpCommand(st, "Current Threads", e);
    }

}

File: chad/src/main/java/dev/shog/chad/commands/developer/CurrentThreads.kt
package dev.shog.chad.commands.developer

import com.sun.management.OperatingSystemMXBean
import dev.shog.chad.framework.handle.MessageHandler
import dev.shog.chad.framework.handle.coroutine.CoroutineManager
import dev.shog.chad.framework.obj.Command
import sx.blah.discord.handle.impl.events.guild.channel.message.MessageEvent
import sx.blah.discord.util.EmbedBuilder
import java.lang.management.ManagementFactory
import java.util.*

/**
 * The current running threads within Chad
 *
 * @author sho, codebasepw
 */
class CurrentThreads : Command.Class {
    override suspend fun run(e: MessageEvent, args: MutableList<String>) {
        // Creates an embed builder and applies a title
        val embedBuilder = EmbedBuilder()
        embedBuilder.withTitle("Current Threads Running")

        // Adds all threads running threads to the stringbuilder, than to the description.
        val stringBuilder = StringBuilder()
        CoroutineManager.instance.users.forEach { (key, `val`) ->
            stringBuilder.append('`')
                    .append(key.name)
                    .append("`: ")
                    .append(`val`)
                    .append('\n')
        }

        // Gets the used ram by the JVM, and the available ram and adds it to the stringbuilder
        stringBuilder.append("\nThe JVM is currently using `")
                .append(Runtime.getRuntime().totalMemory() / 1000 / 1000).append("`mb out of `")
                .append(
                        (ManagementFactory.getOperatingSystemMXBean() as OperatingSystemMXBean).totalPhysicalMemorySize / 1000 / 1000
                ).append("`mb.")

        // Append to builder
        embedBuilder.appendDesc(stringBuilder.toString())

        // Sends
        MessageHandler(e.channel, e.author).sendEmbed(embedBuilder)
    }

    override suspend fun help(e: MessageEvent) {
        val st = HashMap<String, String>()
        st["threads"] = "Displays all running threads for users."
        Command.helpCommand(st, "Current Threads", e)
    }

}
